lib/series_statistics.rb

```diff
@@ -20,6 +20,12 @@ def standard_deviation
     return Math.sqrt(self.variance)
   end
 
+  def median
+    num_array = (self.data.sort.reject{|a| a[1].nil?}).map { |a| a[1]}.sort
+    len = num_array.length
+    (num_array[(len - 1) / 2] + num_array[len / 2]) / 2.0
```

Comment:
There are two issues with this logic.
Take a look at the algorithm here: https://www.oreilly.com/library/view/ruby-cookbook/0596523696/ch02s08.html